### ISSUE TITLE
Bump Axios version?

### ISSUE THREAD

**Issue by gombosg** [Flyrell/axios-auth-refresh#58]
Hi, I tried the new TS supported version but it throws "incompatible types" errors. It's probably, because the axios version is locked to ~0.18.0 but the latest release is 0.19. Maybe unblock this restriction so that npm could use the latest to avoid type conflicts?

```
Argument of type 'import("/node_modules/axios/index").AxiosInstance' is not assignable to parameter of type 'import("/node_modules/axios-auth-refresh/node_modules/axios/index").AxiosInstance'.
  The types of 'defaults.adapter' are incompatible between these types.
    Type 'import("/node_modules/axios/index").AxiosAdapter | undefined' is not assignable to type 'import("/node_modules/axios-auth-refresh/node_modules/axios/index").AxiosAdapter | undefined'.
      Type 'import("/node_modules/axios/index").AxiosAdapter' is not assignable to type 'import("/node_modules/axios-auth-refresh/node_modules/axios/index").AxiosAdapter'.
        Types of parameters 'config' and 'config' are incompatible.
          Type 'import("/node_modules/axios-auth-refresh/node_modules/axios/index").AxiosRequestConfig' is not assignable to type 'import("/node_modules/axios/index").AxiosRequestConfig'.
            Types of property 'method' are incompatible.
              Type 'string | undefined' is not assignable to type '"get" | "GET" | "delete" | "DELETE" | "head" | "HEAD" | "options" | "OPTIONS" | "post" | "POST" | "put" | "PUT" | "patch" | "PATCH" | undefined'.
                Type 'string' is not assignable to type '"get" | "GET" | "delete" | "DELETE" | "head" | "HEAD" | "options" | "OPTIONS" | "post" | "POST" | "put" | "PUT" | "patch" | "PATCH" | undefined'.
```

edit I'm happy to make a PR if we agree on some solution.

**Comment by Flyrell** [comment]
I can't unblock it because of https://github.com/axios/axios/issues/2295. The interceptor wouldn't work properly without the custom request parameters which were deleted in v0.19 of axios. This is a known bug and should be re-added to axios, but as of now, we cannot support axios 0.19

**Comment by gombosg** [comment]
Then how come it is working fine for me on axios 0.19? 

The project may have a dependency on 0.18 but you don't have control over the axios instance version that the library gets.
Also, this is not mentioned in the readme if it is mandatory to use 0.18.

(Or as long as I don't want to skip the interceptor, it will be fine?)

**Comment by Flyrell** [comment]
This has been fixed in [axios/axios#2207](https://github.com/axios/axios/pull/2207), so I guess we don't need to bother any more. Also, I think it will be nice to add axios as an peerDependency instead of dependency.

**Comment by tmilos** [comment]
@Flyrell  Axios v0.19.1 is released and it includes axios/axios#2207. Can we get peerDependency to `~0.18.0 || ~0.19.1`?

**Comment by Flyrell** [comment]
@tmilos Will try to update ASAP
